addons/website/models/ir_qweb.py

```diff
@@ -110,6 +118,49 @@ def _post_processing_att(self, tagName, atts):
         if not website:
             return atts
 
+        if (
+            website.cookies_bar
+            and website.block_third_party_domains
+            and not self.env.context.get('cookies_allowed')
+            and not request.env.user.has_group('website.group_website_restricted_editor')
+        ):
+            # If the cookie banner is activated, 3rd-party embedded iframes and
+            # scripts should be controlled. As such:
+            # - 'domains' is a watchlist on the iframe/script's src itself,
+            # - 'classes' is a watchlist on container elements in which iframes
+            # are/could be built on the fly client-side for some reason.
+            cookies_watchlist = {
+                'domains': website.blocked_third_party_domains.split('\n'),
+                'classes': {
+                    's_map',
+                    's_instagram_page',
+                    'o_facebook_page',
+                    'o_background_video',
+                    'media_iframe_video',
+                },
+            }
+            remove_src = False
+            if tagName in ('iframe', 'script'):
+                src_host = parse_url((atts.get('src') or '').lower()).host
+                if src_host:
+                    remove_src = any(
+                        # "www.example.com" and "example.com" should block both.
+                        src_host.endswith(domain.removeprefix('www.'))
+                        for domain in cookies_watchlist['domains']
+                    )
```

Comment:
Would not this block `www.ilovegoogle.com` by mistake ? Maybe:
- strip www from src_host too
- check for exact match or for "ends with '.' + domain"

?